flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/GenerateUtils.scala

```diff
@@ -375,7 +375,7 @@ object GenerateUtils {
              |  $SQL_TIMESTAMP.fromEpochMillis(${ts.getMillisecond}L, ${ts.getNanoOfMillisecond});
            """.stripMargin
         ctx.addReusableMember(fieldTimestamp)
-        generateNonNullLiteral(literalType, fieldTerm, literalType)
+        generateNonNullLiteral(literalType, fieldTerm, ts)
```

Comment:
Don't mix the planner's single line modification with hive's modification, and there is no test.
You should add a single commit to explain why we need it and what is this bug.

Reply:
@xuefuz @JingsongLi This is a bug in planner when generating timestamp literals: we pass the type instead of the value as the literal. The added test exposes the issue. Do you think this change needs a dedicated PR?

Reply:
I think we don't need to dedicated PR for little changes, but we should a dedicated commit at least.

Reply:
And I remember this literal value is internal object, it should be `SqlTimestamp`.

Reply:
Right, `ts` is a `SqlTimestamp `. I'll squash and make a separate commit for this.

Reply:
Now this change is moved to b5f0206